### PR TITLE
Upgrade app to React Native 50, React 16

### PR DESCRIPTION
**Pull request by carloskelly13**
- [x] Switch to React Navigation Stacks
- [x] Remove old navigation.js code
- [x] Unified `index.ios.js` and `index.android.js` to `index.js`
- [x] Upgrade to React Native 50 and React 16
- [x] Upgrade Gradle build tools
- [x] Fix layout on iPhone X
- [x] Add `yarn.lock` (the main reason this PR is so big in LOC)

**Tested on both simulators and real hardware (iPhone X and Moto G2)**

android
ios-small
ios

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
         android:name=".MainActivity"
         android:label="@string/app_name"
         android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
-        android:screenOrientation="portrait">
```
Comment (jevakallio) [inline android/app/src/main/AndroidManifest.xml:23]: Do we not need portrait orientation locking anymore @carlospaelinck?

File: src/components/example-list.js [hunk 2]
```diff
@@ -35,9 +36,11 @@ export default class ExampleList extends Component {
     });
 
     this.state = {
-      dataSource: ds.cloneWithRows(components.map((component) => {
-        return React.createElement(Example, { module: component });
-      })),
+      dataSource: ds.cloneWithRows(
```
Comment (jevakallio) [inline src/components/example-list.js:39]: Reindentation courtesy of Prettier.

**Review by jevakallio — APPROVED** [review]
:shipit:

**Comment by jpdriver** [comment]
Could be worth re-using those GIFs on the `README.md`? 😬

**Pull request merged**